notebooks/curation_api/python/src/collection.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+import requests
+
+from src.utils.logger import get_custom_logger
+from src.utils.http import url_builder, get_headers
+
+
+logger = get_custom_logger()
+
+
+def create_collection(collection_form_metadata: str) -> str:
+    """
+    Create a new private Collection
+    :param collection_form_metadata: the Collection metadata to use to instantiate a Collection
+    :return: the Collection uuid
+    """
+    url = url_builder("/collections")
+    headers = get_headers()
+    try:
+        res = requests.post(url, data=json.dumps(collection_form_metadata), headers=headers)
+        res.raise_for_status()
+        data = res.json()
+    except Exception as e:
+        logger.error("\n\033[1m\033[38;5;9mFAILED\033[0m")  # 'FAILED' in bold red
+        raise e
+    else:
+        collection_uuid = data.get("collection_uuid")
+        logger.info("\n\033[1m\033[38;5;10mSUCCESS\033[0m\n")  # 'SUCCESS' in bold green
+        logger.info(f"New private Collection uuid:\n{collection_uuid}\n")
+        logger.info(f"New private Collection url:\n{os.getenv('site_url')}/collections/{collection_uuid}")
+        return collection_uuid
+
+
+def create_revision(collection_uuid: str) -> str:
+
+    url = url_builder(f"/collections/{collection_uuid}/revision")
+    headers = get_headers()
+    try:
+        res = requests.post(url, headers=headers)
+        res.raise_for_status()
+        data = res.json()
+    except Exception as e:
+        logger.error("\n\033[1m\033[38;5;9mFAILED\033[0m")  # 'FAILED' in bold red
+        raise e
+    else:
+        revision_uuid = data.get("revision_id")
+        logger.info("\n\033[1m\033[38;5;10mSUCCESS\033[0m\n")  # 'SUCCESS' in bold green
+        logger.info(f"Revision uuid:\n{revision_uuid}\n")
+        logger.info(f"Revision url:\n{os.getenv('site_url')}/collections/{revision_uuid}")
```

Comment:
Nice! This reminds me, we should moving to using [HATEOAS](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#use-hateoas-to-enable-navigation-to-related-resources) in our api responses

Reply:
Related to the links: https://github.com/chanzuckerberg/single-cell-data-portal/issues/2638 is mentioned in Brian's 'Submit Find Manage' doc for the upcoming half.